commons/src/main/java/fr/free/nrw/commons/category/CategorizationFragment.java

```diff
@@ -367,8 +452,28 @@ private void startUpdatingCategoryList() {
         if (lastUpdater != null) {
             lastUpdater.cancel(true);
         }
+
+        if (methodAUpdater != null) {
+            methodAUpdater.cancel(true);
+        }
+
+
+        ArrayList<CategoryItem> itemList = new ArrayList<CategoryItem>(itemSet);
+
+        methodAUpdater = new MethodAUpdater();
         lastUpdater = new CategoriesUpdater();
+
         Utils.executeAsyncTask(lastUpdater, executor);
+        Utils.executeAsyncTask(methodAUpdater, executor);
+
+
+        categoriesAdapter.setItems(itemList);
```

Comment:
Thanks @niedzielski! I just realized that AsyncTask doesn't work like a function, the statements after its call are executed immediately instead of after it returns. But then how do I call a method AFTER onPostExecute()? I've looked around and there doesn't seem to be a clear explanation of where the control goes after the AsyncTasks onPostExecute()s are done.